ada_feeding/ada_feeding/trees/acquire_food_tree.py

```diff
@@ -73,9 +55,69 @@ def create_move_to_tree(
         -------
         tree: The behavior tree that acquires food
         """
+        # TODO: remove name and tree_root_name
+        # We have access to them implicitly via self.blackboard / self.blackboard_tree_root
+
+        ### Inputs we expect on the blackboard on initialization
+        # Note: WRITE access since send_goal could write to these
+        # Mask for ComputeFoodFrame
+        self.blackboard.register_key(key="mask", access=py_trees.common.Access.WRITE)
+        # CameraInfo for ComputeFoodFrame
+        self.blackboard.register_key(
+            key="camera_info", access=py_trees.common.Access.WRITE
+        )
+
+        ### Define Tree Leaves and Subtrees
+
+        # Add ComputeFoodFrame
+        compute_food_frame = ComputeFoodFrame(
+            "ComputeFoodFrame", self.blackboard.namespace
+        )
+        compute_food_frame.blackboard_inputs(
+            ros2_node=node,
+            camera_info=BlackboardKey("camera_info"),
+            mask=BlackboardKey("mask"),
+            # Default world_frame = "world"
+            # Default debug_tf_frame = "food"
+        )
+        compute_food_frame.blackboard_outputs(
+            action_select_request=None,
+            food_frame=None,
+            debug_tf_publisher=BlackboardKey(
+                "debug_tf_publisher"
+            ),  # Set to None to disable
+        )
```

Comment:
Not crucial, but as it stands we have to do 3 function calls to fully define a `BlacboardBehavior`, which is not a common pattern in Python. I think it would be cleaner if `BlackboardBehavior.__init__` takes in `blackboard_inputs` and `blackboard_outputs` as a Dict parameters, and then calls the respective functions to set them (e.g., self.blackboard_inputs(**blackboard_inputs)).

Reply:
Adding dictionaries to __init__ introduces a few of its own problems:

1. It kills per-key static type-checking.
2. It allows a client of the behavior to add read/write keys to the Behavior blackboard without explicit specification by the programmer of the behavior

Reply:
I don’t think it does. I’m proposing still having the functions `blackboard_inputs` and `blackboard_outputs` (to allow static type checking and to prevent programmers from adding undefined keys), but we merely provide a shortcut to initializing them via the __init__ function. Static type checking and ensuring that the right keys are passed will still be done in the `blsckboard_input`/output function